bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/api/RestUtils.java

```diff
@@ -0,0 +1,193 @@
+        MessageDigest digest;
+        try {
+            digest = MessageDigest.getInstance("SHA-512");
+        } catch (NoSuchAlgorithmException e) {
+            logger.error("Definitively, it is unexpected.", e);
+            throw new IllegalStateException("Unexpected error. SHA-512 algorithm must exists in JDK distribution", e);
+        }
+        digest.reset();
+        digest.update(pwdToEnc.getBytes(StandardCharsets.UTF_8));
+
+        return String.format("%0128x", new BigInteger(1, digest.digest()));
+    }
+
+    public static byte[] getOauth2Sig(String messageSign, String secret) {
+        byte[] secretBytes = secret.getBytes(StandardCharsets.UTF_8);
+        SecretKeySpec signingKey = new SecretKeySpec(secretBytes, HMAC_SHA1_ALGORITHM);
+
+        try {
+            Mac mac = Mac.getInstance(HMAC_SHA1_ALGORITHM);
+            mac.init(signingKey);
+            return Base64.getEncoder().encode(mac.doFinal(messageSign.getBytes(StandardCharsets.UTF_8)));
+        } catch (NoSuchAlgorithmException e) {
+            logger.error("Unexpected error. SHA1 algorithm must exists in JDK distribution.", e);
+            throw new IllegalStateException("Unexpected error. SHA1 algorithm must exists in JDK distribution", e);
+        } catch (InvalidKeyException e) {
+            logger.error("Unexpected error.", e);
+            throw new IllegalStateException("Unexpected error.", e);
+        }
+    }
+
+    public static byte[] getTokenSignature(String authUrl, String secretKey, Map<String, String> empData,
+            String timestamp) {
+        UriBuilder builder = UriBuilder.fromUri(authUrl);
+        empData.forEach(builder::queryParam);
+
+        URI reqUri = builder.build();
+        String signUrl = empData.size() > 0 ? reqUri.getPath() + "?" + reqUri.getRawQuery() : reqUri.getPath();
+        String messageToSign = String.format("%s\n%s", signUrl, timestamp);
+        return getOauth2Sig(messageToSign, secretKey);
+    }
+
+    public static RestResult getCall(HttpClient httpClient, String encodedUrl, @Nullable Map<String, String> headers,
+            @Nullable Map<String, String> params) throws IOException {
+
+        Request request = httpClient.newRequest(encodedUrl).method("GET");
+        if (params != null) {
+            params.forEach(request::param);
+        }
+        if (headers != null) {
+            headers.forEach(request::header);
+        }
+
+        if (logger.isTraceEnabled()) {
+            logger.trace("GET request: {}", request.getURI());
+        }
+        try {
+            ContentResponse response = request.send();
+
+            logger.trace("GET response: {}", response.getContentAsString());
+
+            return new RestResult(response.getContentAsString(), response.getStatus());
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.error("Exception occurred during GET execution: {}", e.getMessage(), e);
```

Comment:
Exception is thrown and handled, so no need to log to error
```suggestion
            logger.debug("Exception occurred during GET execution: {}", e.getMessage(), e);
```